src/net/atlas/db.rs

```diff
@@ -73,9 +87,43 @@ const ATLASDB_INITIAL_SCHEMA: &'static [&'static str] = &[
     "CREATE TABLE db_config(version TEXT NOT NULL);",
 ];
 
+const ATLASDB_SCHEMA_2: &'static [&'static str] = &[
+    // We have to allow status to be null, because SQLite won't let us add
+    //  a not null column without a default. The default defeats the point of
+    //  having not-null here anyways, so we leave this field nullable.
+    r#"
+    ALTER TABLE attachment_instances
+    ADD status INTEGER
+    ;"#,
+    // All of the attachment instances that previously existed in the database
+    //  already were "checked", so set status to 2 (which corresponds to "checked").
+    r#"
+    UPDATE attachment_instances SET status = 2;
+    "#,
+    r#"
+    CREATE INDEX IF NOT EXISTS index_instance_status ON attachment_instances(status);
+    "#,
+];
+
 const ATLASDB_INDEXES: &'static [&'static str] =
     &["CREATE INDEX IF NOT EXISTS index_was_instantiated ON attachments(was_instantiated);"];
 
+/// Attachment instances pass through different states once written to the AtlasDB.
+/// These instances are initially written as a new Stacks block is processed, and marked
+/// as `Queued`. These queued instances contain all the data of the new attachment instance,
+/// but they have not yet been checked against the AtlasDB to determine if there is extant
+/// Attachment content/data associated with them. The network run loop (`p2p` thread) checks
+/// for any queued attachment instances on each pass, and performs that check. Once the check
+/// is completed, any checked instances are updated to `Checked`.
+pub enum AttachmentInstanceStatus {
+    /// This variant indicates that the attachments instance has been written,
+    /// but the downloader has not yet checked that the attachment matched
```

Comment:
By "downloader" you mean the "downloader in the p2p thread", right?

Reply:
Yes, the `AttachmentsDownloader`, will update the comment to make more clear

Reply:
Added in https://github.com/stacks-network/stacks-blockchain/pull/3086/commits/cb9f8dca0f9697e7cfe04c1b2e776de4b8aec8ce